R/carehomes.R

```diff
@@ -1810,6 +2066,26 @@ carehomes_particle_filter_data <- function(data) {
     stop("Non-hospital deaths are not consistently split into total vs care
          homes/community")
   }
+  if (## aggregated and stratified pillar 2 positivity data passed by mistake
+    (any(!is.na(data$pillar2_over25_tot)) || (any(!is.na(data$pillar2_tot)))) &&
+    (any(!is.na(data$pillar2_under15_tot)) ||
+     any(!is.na(data$pillar2_15_24_tot)) ||
+     any(!is.na(data$pillar2_25_49_tot)) ||
+     any(!is.na(data$pillar2_50_64_tot)) ||
+     any(!is.na(data$pillar2_65_79_tot)) ||
+     any(!is.na(data$pillar2_80_plus_tot)))) {
+    stop("Cannot double fit age-specific and aggregated pillar 2 positivity!")
+  }
+  if (## aggregated and stratified pillar 2 cases data passed by mistake
+    (any(!is.na(data$pillar2_over25_cases)) || (any(!is.na(data$pillar2_cases)))) &&
+    (any(!is.na(data$pillar2_under15_cases)) ||
+     any(!is.na(data$pillar2_15_24_cases)) ||
+     any(!is.na(data$pillar2_25_49_cases)) ||
+     any(!is.na(data$pillar2_50_64_cases)) ||
+     any(!is.na(data$pillar2_65_79_cases)) ||
+     any(!is.na(data$pillar2_80_plus_cases)))) {
+    stop("Cannot double fit age-specific and aggregated pillar 2 cases!")
+  }
```

Comment:
These look good but I'm not sure these work for the multiregion version. I guess they need to be written in a way like check_pillar2_streams below

Reply:
Also it should be possible to fit under15 and/or 15_24 alongside over25

Reply:
We also want to prevent double fitting via mixing, e.g. fitting over25 positivity and 65_79 cases. Maybe we just enforce that you can't fit any mix of positivity and cases?

Reply:
had another stab at it; see what you think